features/src/test/scala/com/salesforce/op/features/FeatureSparkTypeTest.scala

```diff
@@ -30,70 +30,84 @@
 class FeatureSparkTypeTest extends FlatSpec with TestSparkContext {
-  val sparkTypeToTypeTagMappings = Seq(
-    (DoubleType, weakTypeTag[types.RealNN]), (FloatType, weakTypeTag[types.RealNN]),
-    (LongType, weakTypeTag[types.Integral]), (IntegerType, weakTypeTag[types.Integral]),
-    (ShortType, weakTypeTag[types.Integral]), (ByteType, weakTypeTag[types.Integral]),
-    (DateType, weakTypeTag[types.Date]), (TimestampType, weakTypeTag[types.DateTime]),
-    (StringType, weakTypeTag[types.Text]), (BooleanType, weakTypeTag[types.Binary]),
-    (VectorType, weakTypeTag[types.OPVector])
+  val primitiveTypes = Seq(
+    (DoubleType, weakTypeTag[types.Real], DoubleType),
+    (FloatType, weakTypeTag[types.Real], DoubleType),
+    (LongType, weakTypeTag[types.Integral], LongType),
+    (IntegerType, weakTypeTag[types.Integral], LongType),
+    (ShortType, weakTypeTag[types.Integral], LongType),
+    (ByteType, weakTypeTag[types.Integral], LongType),
+    (DateType, weakTypeTag[types.Date], LongType),
+    (TimestampType, weakTypeTag[types.DateTime], LongType),
+    (StringType, weakTypeTag[types.Text], StringType),
+    (BooleanType, weakTypeTag[types.Binary], BooleanType),
+    (VectorType, weakTypeTag[types.OPVector], VectorType)
   )
 
-  val sparkCollectionTypeToTypeTagMappings = Seq(
-    (ArrayType(LongType, containsNull = true), weakTypeTag[types.DateList]),
-    (ArrayType(DoubleType, containsNull = true), weakTypeTag[types.Geolocation]),
-    (MapType(StringType, StringType, valueContainsNull = true), weakTypeTag[types.TextMap]),
-    (MapType(StringType, DoubleType, valueContainsNull = true), weakTypeTag[types.RealMap]),
-    (MapType(StringType, LongType, valueContainsNull = true), weakTypeTag[types.IntegralMap]),
-    (MapType(StringType, BooleanType, valueContainsNull = true), weakTypeTag[types.BinaryMap]),
-    (MapType(StringType, ArrayType(StringType, containsNull = true), valueContainsNull = true),
-      weakTypeTag[types.MultiPickListMap]),
-    (MapType(StringType, ArrayType(DoubleType, containsNull = true), valueContainsNull = true),
-      weakTypeTag[types.GeolocationMap])
+  val nonNullable = Seq(
+    (DoubleType, weakTypeTag[types.RealNN], DoubleType),
+    (FloatType, weakTypeTag[types.RealNN], DoubleType)
   )
 
-  val sparkNonNullableTypeToTypeTagMappings = Seq(
-    (DoubleType, weakTypeTag[types.Real]), (FloatType, weakTypeTag[types.Real])
+  private def mapType(v: DataType) = MapType(StringType, v, valueContainsNull = true)
+  private def arrType(v: DataType) = ArrayType(v, containsNull = true)
+
+  val collectionTypes = Seq(
+    (arrType(LongType), weakTypeTag[types.DateList], arrType(LongType)),
+    (arrType(DoubleType), weakTypeTag[types.Geolocation], arrType(DoubleType)),
+    (arrType(StringType), weakTypeTag[types.TextList], arrType(StringType)),
+    (mapType(StringType), weakTypeTag[types.TextMap], mapType(StringType)),
+    (mapType(DoubleType), weakTypeTag[types.RealMap], mapType(DoubleType)),
```

Comment:
do you need map float, map int, map short, etc as well?

Reply:
I thought about it, but I dont think it's that important. Maybe in another PR.